bsip-0086.md

```diff
@@ -0,0 +1,115 @@
+    BSIP: 0086
+    Title: Share market fees to the network
+    Author: Abit More <https://github.com/abitmore>
+    Status: Draft
+    Type: Protocol
+    Created: 2019-11-29
+    Discussion: https://github.com/bitshares/bsips/issues/194
+    Worker: TBD
+
+# Abstract
+
+This BSIP proposes a protocol change: when every trade happens, if there are
+market fees generated, a portion (E.G. 5% or 10%) of the market fees goes to
+the network.
+
+# Motivation
+
+The network needs more tools to generate income to support its development.
+
+# Rationale
+
+A major activity in the network is trading assets. The asset owners make
+profits in the form of market trading fees. Currently, the main tools for the
+network to generate income from asset trading activities are order creation
+fee and order cancellation fee. It seems reasonable that the asset owners
+share some profits (a part of market fees) to the network to support its
+development.
+
+The cut of market fees can go to committee-account's vesting balances.
+As a supporting measure, committee-account should be exempted from
+white-listing restrictions, so that it is able to claim the vesting balances
+and sell them for core token or other tokens to pay for development later.
```

Comment:
I think it would make sense to add the following:

"The committee is instructed to regularly (at least once every quarter) convert incoming market fees into BitShares native assets: BTS (favorable) or any BitAsset."

Reply:
I don't think it's good to say that the committee should do this or that in the document.

Reply:
I don't think it's good to give an entity a role if that entity does not accept those responsibilities explicitly. Then we should think of a new entity to be the receiver.

Reply:
?

Reply:
IMO the scope of a BSIP should be as small as needed. It's hard to define how to efficiently use the potential funds right now. Selling the funds for BTS or bitAsset is not the only way to use the funds. For example, it could be easier for the committee to directly pay gateway.BTC for development. I'd rather let the committee decide when they see fit.

Reply:
> IMO the scope of a BSIP should be as small as needed. It's hard to define how to efficiently use the potential funds right now. Selling the funds for BTS or bitAsset is not the only way to use the funds. For example, it could be easier for the committee to directly pay gateway.BTC for development. I'd rather let the committee decide when they see fit.

Considering the debacle with CryptoBridge holding any value in native assets should be preferrable. I see the point that a BSIP can't and shouldn't tell committee what to do. The very least ALL current active committee members should confirm their willingness for taking on that responsibility. This might be considered unfair, with the information available to me my personal trust regarding committee towards workers and transparent payments is not very high at the moment, considering what has happened in the past with worker proposals governed by committee.

Reply:
Another thought: 
How about the following

a) let the committee decide who will be the `market_fee_network_budget_account` of the incoming budget via committee parameter (value is an account name). This may be very well just the committee-account in the beginning.
b) if desirable, allow a fee for the one that handles the budget. To do that, define additionally a `market_fee_network_budget_handler` (value is an account name) and a `market_fee_network_budget_handler_fee` (value is percent). Then the incoming budget goes `1-market_fee_network_budget_handler_fee`% to the `market_fee_network_budget_account` and `market_fee_network_budget_handler_fee`% to the  `market_fee_network_budget_handler`. The handler should have access to the `market_fee_network_budget_account` account of course, but this is external configuration of account permissions and not a core requirement (would e.g. allow use of BSIP 40 permissions)

Reply:
@bitcrab see here

Reply:
"market_fee_network_budget_handler" complicates things related to white-listing.

Reply:
> "market_fee_network_budget_handler" complicates things related to white-listing. IMO it's unnecessary.

Not more than if the asset owner would use market fee sharing, same logic.

Reply:
> "market_fee_network_budget_handler" complicates things related to white-listing. IMO it's unnecessary.

agree, it bring more complex unnecessarily.

Reply:
> Not more than if the asset owner would use market fee sharing, same logic. Or do I miss something?

If an asset owner would use market fee sharing, she will consider the matters about whitelisting. The situation here is different, here the committee is doing something but not the asset owners. By using the committee-account directly, we only need to exclude that one account from whitelisting restrictions; by using a dynamic account we need more complicated logic.

Reply:
True, complexity would be increased. Still, a strong referral rewards program is exactly what other exchanges have that makes them grow massively.

Reply:
Okay, if the committee is accepting this role pro-bono ...